render/gles2/renderer.c

```diff
@@ -317,8 +317,8 @@ static bool gles2_read_pixels(struct wlr_renderer *wlr_renderer,
 	} else {
 		// Unfortunately GLES2 doesn't support GL_PACK_*, so we have to read
 		// the lines out row by row
-		for (size_t i = src_y; i < src_y + height; ++i) {
-			glReadPixels(src_x, src_y + height - i - 1, width, 1, fmt->gl_format,
+		for (size_t i = 0; i < height; ++i) {
+	        glReadPixels(src_x, renderer->viewport_height - src_y - i, width, 1, fmt->gl_format,
```

Comment:
If `src_y = 0` and `i = 0`, then `renderer->viewport_height - src_y - i == renderer->viewport_height`. This is off-by-one for a y coordinate.